emperor/support_files/js/scale-view-controller.js

```diff
@@ -0,0 +1,278 @@
+        }
+        scope.resize();
+        var attributes = scope.getScale(uniqueVals, scaled);
+
+        // fetch the slickgrid-formatted data
+        var data = decompViewDict.setCategory(
+          attributes, scope.setPlottableAttributes, category);
+
+        scope.setSlickGridDataset(data);
+      },
+      'slickGridColumn': {id: 'title', name: '', field: 'value',
+        sortable: false, maxWidth: SLICK_WIDTH,
+        minWidth: SLICK_WIDTH,
+        autoEdit: true,
+        //formatter: Slick.Formatters.Checkmark,
+        editor: ScaleEditor.ScaleEditor}};
+
+    EmperorAttributeABC.call(this, container, title, helpmenu,
+        decompViewDict, options);
+    this.$header.append(this.$scaledValue);
+    this.$header.append(this.$scaledLabel);
+    this.$body.prepend(this.$globalDiv);
+
+    scope.$scaledValue.on('change', options.categorySelectionCallback);
+
+    return this;
+  }
+  ScaleViewController.prototype = Object.create(EmperorAttributeABC.prototype);
+  ScaleViewController.prototype.constructor = EmperorAttributeABC;
+
+  /**
+   * Converts the current instance into a JSON string.
+   *
+   * @return {Object} JSON ready representation of self.
+   */
+  ScaleViewController.prototype.toJSON = function() {
+    var json = EmperorAttributeABC.prototype.toJSON.call(this);
+    var sliderVal = this.$globalDiv.children('input').val();
+    json.globalScale = sliderVal;
+    json.scaleVal = this.$scaledValue.is(':checked');
+    return json;
+  };
+
+  /**
+   * Decodes JSON string and modifies its own instance variables accordingly.
+   *
+   * @param {Object} Parsed JSON string representation of self.
+   */
+  ScaleViewController.prototype.fromJSON = function(json) {
+    // Can't call super because select needs to be set first
+    // Order here is important. We want to set all the extra controller
+    // settings before we load from json, as they can override the JSON when set
+    this.$select.val(json.category);
+    this.$select.trigger('chosen:updated');
+    this.$sliderGlobal.slider('value', json.globalScale);
+    this.$scaledValue.prop('checked', json.scaleVal);
+    this.$scaledValue.trigger('change');
+
+    // fetch and set the SlickGrid-formatted data
+    var k = this.getActiveDecompViewKey();
+    var data = this.decompViewDict[k].setCategory(
+      json.data, this.setPlottableAttributes, json.category);
+    this.setSlickGridDataset(data);
+    // set all to needsUpdate
+    this.decompViewDict[k].needsUpdate = true;
+  };
+
+  /**
+   * Resizes the container and the individual elements.
+   *
+   * Note, the consumer of this class, likely the main controller should call
+   * the resize function any time a resizing event happens.
+   *
+   * @param {float} width the container width.
+   * @param {float} height the container height.
+   */
+  ScaleViewController.prototype.resize = function(width, height) {
+    this.$body.height(this.$canvas.height() - this.$header.height());
```

Comment:
Any reason why the body needs to be resized here? It gets resized again when the `super.resize` is getting called.